syntax="proto3";
package eos.console;
import "Acl.proto";
import "Ns.proto";
import "Find.proto";
import "Fs.proto";
import "Rm.proto";
import "StagerRm.proto";
import "Route.proto";
import "Recycle.proto";
import "Io.proto";
import "Group.proto";
import "Debug.proto";
import "Node.proto";
import "Quota.proto";
import "Space.proto";
import "Config.proto";
import "Access.proto";
import "Fsck.proto";
import "Token.proto";
import "QoS.proto";
import "Convert.proto";
import "Share.proto";

//------------------------------------------------------------------------------
// Request message sent to the server
//------------------------------------------------------------------------------
message RequestProto {
  // Types of output format supported
  enum FormatType {
    DEFAULT = 0;
    JSON    = 1;
    HTTP    = 2;
    FUSE    = 3;
  }

  FormatType format = 1;

  // Actual request data object
  oneof command {
    AclProto acl           = 2;
    NsProto  ns            = 3;
    //    DrainProto drain       = 4; @note not used, can be reused later on
    FindProto find         = 5;
    FsProto fs             = 6;
    RmProto rm             = 7;
    StagerRmProto stagerRm = 8;
    RouteProto route       = 9;
    RecycleProto recycle   = 10;
    IoProto io             = 11;
    GroupProto group       = 12;
    DebugProto debug       = 13;
    NodeProto node         = 14;
    QuotaProto quota       = 15;
    SpaceProto space       = 16;
    ConfigProto config     = 17;
    AccessProto access     = 18;
    FsckProto fsck         = 19;
    TokenProto token       = 20;
    QoSProto qos           = 21;
    ConvertProto convert   = 22;
    ShareProto share       = 23;
  }

  string Comment = 30;
  bool DontColor = 31; // True if the output fd is not a tty.
}
